### PR TITLE
[dagster-airlift] mwaa example

### PR DESCRIPTION
**Pull request by dpeng817**
Adds example code for having dags running in mwaa and observing them running locally in dagster.

The goal here is twofold:
- Provide sample code for using the mwaa utility functions we build
- Provide an easily stood up and reproducible test bed for deploying code to mwaa and testing underlying dagster functionality.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23574?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23574** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23574?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23555** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23555?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23402** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23402?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23386** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23386?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23370** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23370?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23493** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23493?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23538** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23538?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23532** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23532?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23512** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23512?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23492** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23492?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23487** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23487?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23483** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23483?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23481** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23481?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23480** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23480?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23444** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23444?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/mwaa/auth.py [hunk 1]
```diff
@@ -31,16 +32,17 @@ def get_session_info(region: str, env_name: str) -> Tuple[str, str]:
 
 
 class MwaaSessionAuthBackend(AirflowAuthBackend):
-    def __init__(self, region: str, env_name: str):
+    def __init__(self, region: str, env_name: str, profile_name: Optional[str] = None):
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/mwaa/auth.py:35]: Hmmm maybe we should just take in the `boto.Session` object directly as we might just replicate the entire interface here.

File: examples/experimental/dagster-airlift/examples/mwaa-example/scripts/setup_mwaa.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+from typing import List
+
+import boto3
+import click
+from botocore.exceptions import ClientError
+
+
+def get_s3_objects(s3_client: boto3.client, bucket: str, prefix: str) -> List[dict]:
+    """Get list of objects in the S3 bucket with the given prefix."""
+    try:
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/examples/mwaa-example/scripts/setup_mwaa.py:20]: This is a lot of code. Does MWAA not provide some of this stuff as utilities? This looks like we are reimplementing AWS tooling with ChatGPT or something.

**Review by schrockn — CHANGES_REQUESTED** [review]
Inline comments

File: examples/experimental/dagster-airlift/examples/mwaa-example/scripts/setup_mwaa.py (hunk 2 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/examples/mwaa-example/scripts/setup_mwaa.py:20]: I ripped this straight from the aws documentation; didn't use chat GPT for this

File: examples/experimental/dagster-airlift/dagster_airlift/mwaa/auth.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/mwaa/auth.py:35]: I think we should just revisit the interface in general. I think we should be aiming to only auth the mwaa session once in general, not just retrieving the client I think.

Comment (schrockn) [reply examples/experimental/dagster-airlift/dagster_airlift/mwaa/auth.py:35]: kk

**Review by schrockn — APPROVED** [review]

**Pull request merged**